src/beamlime/logging/handlers.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from logging import FileHandler, Formatter, StreamHandler
+
+from .formatters import BeamlimeAnsiColorFormatter, BeamlimeFormatter, _HeaderFormatter
+from .records import BeamlimeColorLogRecord, BeamlimeLogRecord
+
+
+class _HeaderMixin:
+    def add_header(self):
+        try:
+            msg = self.formatter.header
+            stream = self.stream
+            stream.write(msg + self.terminator)
+            self.flush()
+        except RecursionError:
+            raise
+
+    def setFormatter(self, fmt: _HeaderFormatter | Formatter | None) -> None:
+        super().setFormatter(fmt)
+        if hasattr(fmt, "header") and self.header:
+            self.add_header()
+
+
+class BeamlimeFileHandler(_HeaderMixin, FileHandler):
+    def __init__(
+        self,
+        filename: str,
+        mode: str = "a",
+        encoding: str | None = None,
+        delay: bool = False,
+        errors: str | None = None,
+        header: bool = True,
+    ) -> None:
+        import logging
+
+        if not issubclass(logging.getLogRecordFactory(), BeamlimeLogRecord):
+            logging.setLogRecordFactory(BeamlimeLogRecord)
```

Comment:
Does this work with 3rd party code which logs?

Reply:
It should, if they use `logging.LogRecord`, since the BeamlimeLogRecord is a subclass of LogRecord and it has default values in it.

Reply:
I decided not to change global `LogRecordFactory` and use more general form of log-call instead.